consensus/parlia/parlia.go

```diff
@@ -874,6 +885,19 @@ func (p *Parlia) Seal(chain consensus.ChainHeaderReader, block *types.Block, res
 	return nil
 }
 
+func (p *Parlia) shouldWaitForCurrentBlockProcess(chain consensus.ChainHeaderReader, header *types.Header, snap *Snapshot) bool {
+	highestVerifiedHeader := chain.GetHighestVerifiedHeader()
+	if highestVerifiedHeader == nil || highestVerifiedHeader.Number == nil {
+		return false
+	}
+
+	if !snap.inturn(p.val) && header.Number.Cmp(highestVerifiedHeader.Number) == 0 {
```

Comment:
We need to change the condition here. 
`header.Number.Cmp(highestVerifiedHeader.Number) == 0 && snap.inturn(highestVerifiedHeader.Coinbase)`